src/pages/Facility/services/inventory/externalSupply/deliveryOrder/SmartExternalDeliveryRow.tsx

```diff
@@ -0,0 +1,1040 @@
+        );
+        if (taxes?.length) {
+          form.setValue(getItemFieldPath(index, "tax_components"), taxes);
+        }
+
+        // Extract discounts
+        const discounts =
+          product.charge_item_definition.price_components?.filter(
+            (c) => c.monetary_component_type === MonetaryComponentType.discount,
+          );
+        if (discounts?.length) {
+          form.setValue(
+            getItemFieldPath(index, "discount_components"),
+            discounts,
+          );
+        }
+      } else {
+        form.setValue(`items.${index}.unit_price`, 0);
+      }
+      form.setValue(`items.${index}.is_manually_edited`, false);
+      setIsCreatingNew(false);
+    },
+    [form, index],
+  );
+
+  const handleProductSelect = (product: ProductRead) => {
+    fillFormFromProduct(product);
+    setBatchSelectorOpen(false);
+  };
+
+  const markAsEdited = () => {
+    form.setValue(`items.${index}.is_manually_edited`, true);
+    form.setValue(`items.${index}.supplied_item`, undefined);
+    setIsCreatingNew(true);
+  };
+
+  // Calculate computed total
+  const computedTotal = useMemo(() => {
+    const basePrice = unitPrice || 0;
+    let total = basePrice * (quantity || 1);
+
+    // Apply taxes
+    if (taxComponents?.length) {
+      taxComponents.forEach((tax) => {
+        if (tax.factor) {
+          total += basePrice * (quantity || 1) * (tax.factor / 100);
+        } else if (tax.amount) {
+          total += parseFloat(tax.amount);
+        }
+      });
+    }
+
+    // Apply discounts
+    if (discountComponents?.length) {
+      discountComponents.forEach((discount) => {
+        if (discount.factor) {
+          total -= basePrice * (quantity || 1) * (discount.factor / 100);
+        } else if (discount.amount) {
+          total -= parseFloat(discount.amount);
+        }
+      });
+    }
+
+    return total;
+  }, [unitPrice, quantity, taxComponents, discountComponents]);
+
+  // Available components from facility
+  const availableTaxes = useMemo(
+    () =>
+      (facilityData?.instance_tax_monetary_components ||
+        []) as MonetaryComponentRead[],
+    [facilityData],
+  );
+
+  const availableDiscounts = useMemo(
+    () =>
+      [
+        ...(facilityData?.discount_monetary_components || []),
+        ...(facilityData?.instance_discount_monetary_components || []),
+      ].map((component) => ({
+        ...component,
+        amount:
+          component?.amount != null
+            ? String(component.amount)
+            : component.amount,
+      })) as MonetaryComponentRead[],
+    [facilityData],
+  );
+
+  const getExpirationDisplay = (product: ProductRead) => {
+    return product.expiration_date
+      ? format(new Date(product.expiration_date), "MMM yyyy")
+      : "N/A";
+  };
+
+  return (
+    <TableRow className="divide-x divide-gray-200 hover:bg-gray-50/50">
+      {/* Product Knowledge */}
+      <TableCell className="align-top p-2">
+        <FormField
```

Comment:
Shouldn't all these fields be editable only if we are selecting a new batch? Fields shouldn't be changing for an existing batch no?